File: 프로그래머스/1/82612. 부족한 금액 계산하기/부족한 금액 계산하기.java
class Solution {
    public long solution(int price, int money, int count) {
        
        long total = 0; // 놀이기구 이용 금액
        for(int i=1; i<=count; i++) {
            total += (long) i * price;
        }
        
        long result = total - money;
        
        if (result < 0) {
            return 0;
        }

        return result;
    }
}

File: 프로그래머스/1/82612. 부족한 금액 계산하기/부족한 금액 계산하기.cpp
using namespace std;

long long solution(int price, int money, int count)
{
    long long answer = 0;
    long long result = 0;
    for(int i=1; i<=count; i++) {
        answer += price * i;
    }
    
    if(money - answer < 0) {
        result = answer - money;
    } else {
        result = 0;
    }
    
    return result;
}